x-pack/plugins/triggers_actions_ui/public/types.ts

```diff
@@ -173,6 +177,7 @@ export interface ActionParamsProps<TParams> {
   isLoading?: boolean;
   isDisabled?: boolean;
   showEmailSubjectAndMessage?: boolean;
+  executionMode?: ActionConnectorMode;
```

Comment:
I think it is better if the `executionMode` is required and the framework will pass either `ActionConnectorMode.Test` or `ActionConnectorMode.NORMAL (or similar)` to avoid confusion in the future in case we need to add more execution modes. Another option is to change the `executionMode` to `testingMode?: boolean`.

Reply:
Good point, I'll switch it to being required and create a new enum for the ActionForm code path 👍

Reply:
There's one place where the email connector is instantiated outside of the action framework: [x-pack/plugins/synthetics/public/legacy_uptime/components/settings/default_email.tsx](https://github.com/elastic/kibana/blob/main/x-pack/plugins/synthetics/public/legacy_uptime/components/settings/default_email.tsx#L75)

After chatting with @ymao1 it seemed easiest to keep `executionMode` as optional that way that instance doesn't need to pass a value for no reason. I still changed the location in the action form to pass in the `ActionForm` enum.

Reply:
Ok!